plutus-ledger-constraints/src/Ledger/Constraints/OffChain.hs

```diff
@@ -752,13 +750,12 @@ resolveScriptTxOut
     => ChainIndexTxOut -> m (Maybe ((ValidatorHash, Validator, Language), (DatumHash, Datum), Value))
 resolveScriptTxOut
         Tx.ScriptChainIndexTxOut
-            { Tx._ciTxOutValidator = (vh, v)
+            { Tx._ciTxOutValidator = (vh, _)
             , Tx._ciTxOutScriptDatum = (dh, d)
             , Tx._ciTxOutValue
             } = do
-    -- first check in the 'ChainIndexTx' for the validator, then
-    -- look for it in the 'slOtherScripts map.
-    (validator, pv) <- maybe (lookupValidator vh) (pure . (, PlutusV1)) v
+    -- Look for the validator in the 'slOtherScripts map so we also get the language.
+    (validator, pv) <- lookupValidator vh
```

Comment:
Would it be benificial to add the PlutusVersion alongside the validator in `ChainIndexTxOut`?

Reply:
Just looked at that, it doesn't seem to be possible, at least not easily. We'd also have to store the language in the chain-index.

Reply:
Yeah... quickly looking at cardano-api and ledger, seems like some work and should probably be part of another story. I'm not sure how valuable it is. What do you think?

Reply:
I won't be surprised that we find out soon that we'll have to do it. I'm fine with waiting until that happens.